Lib/http/server.py

```diff
@@ -726,9 +800,36 @@ def send_head(self):
 
             self.send_response(HTTPStatus.OK)
             self.send_header("Content-type", ctype)
-            self.send_header("Content-Length", str(fs[6]))
-            self.send_header("Last-Modified", 
+            self.send_header("Last-Modified",
                 self.date_time_string(fs.st_mtime))
+
+            # Use HTTP compression (gzip) if possible
+            accept_encoding = self.headers.get("Accept-Encoding", "")
+
+            encodings = []
+            for accepted in accept_encoding.split(","):
+                encoding, *quality = accepted.lower().split(";")
+                if quality:
+                    q, *v = quality[0].split("=")
+                    if q == "q" and v and v[0] == "0":
+                        encoding = ""
+                encodings.append(encoding if encoding != "*" else "gzip")
+
+            if ctype in self.compressed_types and "gzip" in encodings:
```

Comment:
_x-gzip_ is the proper name in HTTP 1.0 (which is what the class uses by default), and it is also supposed to be an alias in HTTP 1.1.